ion/symboltoken_test.go

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func newString(value string) *string {
+	return &value
+}
+
+var boolEqualsTestData = []struct {
+	text1   *string
+	sid1    int64
+	source1 *ImportSource
+	text2   *string
+	sid2    int64
+	source2 *ImportSource
+}{
+	{nil, 123, nil,
+		nil, 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 1)},
+	{newString("text1"), 123, nil,
+		newString("text1"), 123, nil},
+	{newString("text2"), 123, newSource("table", 1),
+		newString("text2"), 123, newSource("table", 1)},
+}
+
+func TestBoolEqualsOperator(t *testing.T) {
+	for _, testData := range boolEqualsTestData {
+		st1 := SymbolToken{Text: testData.text1, LocalSID: testData.sid1, Source: testData.source1}
+		st2 := SymbolToken{Text: testData.text2, LocalSID: testData.sid2, Source: testData.source2}
+
+		if !st1.Equal(&st2) {
+			t.Errorf("expected %v, got %v", true, false)
+		}
+	}
+}
+
+var boolNotEqualsTestData = []struct {
+	text1   *string
+	sid1    int64
+	source1 *ImportSource
+	text2   *string
+	sid2    int64
+	source2 *ImportSource
+}{
+	{nil, 123, newSource("table", 1),
+		nil, 123, nil},
+	{nil, 123, nil,
+		newString("text1"), 123, nil},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table2", 1)},
+	{nil, 123, newSource("table", 1),
+		nil, 123, newSource("table", 2)},
+	{newString("text2"), 123, nil,
+		newString("text3"), 123, nil},
+}
```

Comment:
similar comment here about indentation and test cases

Reply:
Updated.